src/crc32.c

```diff
@@ -125,27 +125,43 @@ bool generic_crc32(target_s *const target, uint32_t *const result, const uint32_
 		}
 		const size_t read_len = MIN(sizeof(bytes), len - offset);
 		if (target_mem_read(target, bytes, base + offset, (read_len + 3U) & ~3U)) {
-			DEBUG_ERROR("generic_crc32 error around address 0x%08" PRIx32 "\n", (uint32_t)(base + offset));
+			DEBUG_ERROR("%s: error around address 0x%08" PRIx32 "\n", __func__, (uint32_t)(base + offset));
 			return false;
 		}
 
 		for (size_t i = 0; i < read_len; i++)
 			crc = crc32_calc(crc, bytes[i]);
 	}
-	DEBUG_WARN("%" PRIu32 " ms\n", platform_time_ms() - start_time);
+#ifndef DEBUG_INFO_IS_NOOP
+	/* "generic_crc32: 08000110+75272 {tab} -> 1353 ms,   54 KiB/s" */
+	const uint32_t end_time = platform_time_ms();
+	const uint32_t time_elapsed = end_time - start_time;
+	DEBUG_INFO("%s: %08" PRIx32 "+%" PRIu32 "\t-> %4" PRIu32 " ms", __func__, base, (uint32_t)len, time_elapsed);
+	if (len >= 512U) {
+		const uint32_t speed = len * 1000U / time_elapsed / 1024U;
+		DEBUG_INFO(", %4" PRIu32 " KiB/s", speed);
+	}
```

Comment:
We know why you've added all the whitespace here, but please can we not - this makes it extremely difficult for those of us with line tracking problems to read. It might "look nice" (subjective, no comment here) but that doesn't help us when our eyes keep trying to skip between lines.

```suggestion
	DEBUG_INFO("%s: %08" PRIx32 "+%" PRIu32 " -> %" PRIu32 "ms", __func__, base, (uint32_t)len, time_elapsed);
	if (len >= 512U) {
		const uint32_t speed = len * 1000U / time_elapsed / 1024U;
		DEBUG_INFO(", %" PRIu32 " KiB/s", speed);
	}
```

Would be our strong preference owing to ease of scanning along the line this way without finding we're accidentally reading the next and getting real confused.

Reply:
Reverted/readjusted format-strings to comply. I don't feel strongly about this patch hunk, and I will rarely stare at its output myself once I'm done enough with all the perf dances. Also updated the oneline comment above depicting example output as seen in live testing.